### ISSUE TITLE
Multiple "line" markers become color of last "line" marker

### ISSUE THREAD

**Issue by GoogleCodeExporter** [josejaner/flot#206]
```
I have a graph which I want to look like foo.png. What I found what I was
getting something similar to bar.png. After stepping through the code with
Firebug, I found that listing multiple "line" markers (markers with the
same start and end dates) in a row in the markers section of the options
object will color over all previous markers with the color listed for the
last line marker.

The relevant section of code I use for this is:
    // bp_start_date is a date that occurs before oos_date on the graph
    grid:   { hoverable:  true,
              markings:   [{ xaxis: { from:   bp_start_date, 
                                      to:     oos_date },
                             color:  "#AAF" },
                           { xaxis: { from:   oos_date, 
                                      to:     oos_date },
                             color:  "#FF0000" },
                           { xaxis: { from:   bp_start_date, 
                                      to:     bp_start_date },
                             color:  "#00FF00" }]}
where the free variables are defined correctly elsewhere.

I am no canvas ninja, but the fix I am using is to change lines 913-922 in
jquery.flot.js from:
  if (xrange.from == xrange.to || yrange.from == yrange.to) {
    // draw line
    ctx.strokeStyle = m.color || options.grid.markingsColor;
    ctx.lineWidth = m.lineWidth || options.grid.markingsLineWidth;
    ctx.moveTo(Math.floor(xrange.from), Math.floor(yrange.from));
    ctx.lineTo(Math.floor(xrange.to), Math.floor(yrange.to));
    ctx.stroke();
  }
to:
  if (xrange.from == xrange.to || yrange.from == yrange.to) {
    // draw line
    ctx.beginPath();
    ctx.strokeStyle = m.color || options.grid.markingsColor;
    ctx.lineWidth = m.lineWidth || options.grid.markingsLineWidth;
    ctx.moveTo(Math.floor(xrange.from), Math.floor(yrange.from));
    ctx.lineTo(Math.floor(xrange.to), Math.floor(yrange.to));
    ctx.stroke();
    ctx.closePath();
  }

If that is the proper solution, wohoo! 
Thanks for the awesome library.
```

Original issue reported on code.google.com by `john.mendelewski` on 19 Aug 2009 at 2:18

Attachments:
- [foo.png](https://storage.googleapis.com/google-code-attachments/flot/issue-206/comment-0/foo.png)
- [bar.png](https://storage.googleapis.com/google-code-attachments/flot/issue-206/comment-0/bar.png)

**Comment by GoogleCodeExporter** [comment]
```
Apparently this is fixed in svn, but not as of version 0.5, the featured 
download.
```

Original comment by `john.mendelewski` on 20 Aug 2009 at 3:28

**Comment by GoogleCodeExporter** [comment]
```
Yes, it'll be in the next release. Thanks for the report anyway.
```

Original comment by `olau%iol...@gtempaccount.com` on 21 Oct 2009 at 10:43
- Changed state: **Fixed**
